src/api/app/models/review.rb

```diff
@@ -39,6 +40,8 @@ class NotFoundError < APIException
     end
   end
 
+  before_validation :set_entity_association
```

Comment:
Doesn't this mean that the entity will only be set if you call `review.valid?`? What happens if you dont call .valid? like here?
```ruby
review = Review.new(by_user: user)
review.save
```

I think this should be on `before_save` instead?